interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,956 @@
+            # Lennard-Jones collision diameter in angstroms,
+            # dipole moment in Debye,
+            # polarizability in cubic angstroms, and
+            # rotational relaxation collision number at 298 K.
+            species_string = (
+                f"{species.name:<16}"
+                + f"{geometry[species.transport.geometry]:>4}"
+                + f"{(species.transport.well_depth / ct.boltzmann):>10.3f}"
+                + f"{(species.transport.diameter * 1e10):>10.3f}"
+                + f"{(species.transport.dipole / DEBYE_CONVERSION):>10.3f}"
+                + f"{(species.transport.polarizability * 1e30):>10.3f}"
+                + f"{species.transport.rotational_relaxation:>10.3f}"
+                + "\n"
+            )
+
+            trans_file.write(species_string)
+
+
+def write(
+    solution,
+    output_path=None,
+    reorder_reactions=False,
+    skip_thermo=False,
+    separate_thermo_file=False,
+    skip_transport=False,
+):
+    """Writes Cantera solution object to Chemkin-format file.
+    Parameters
+    ----------
+    solution : cantera.Solution
+        Model to be written
+    output_path : path or str, optional
+        Path of file to be written; if not provided, use cd / 'solution.name'
+    reorder_reactions : bool, optional
+        Flag to reorder reactions based upon molecular weight
+    skip_thermo : bool, optional
+        Flag to skip writing thermo data
+    separate_thermo_file : bool, optional
+        Flag to write thermo data in a seperate file from mechanism
+    skip_transport : bool, optional
+        Flag to skip writing transport data in separate file
+    Returns
+    -------
+    output_file_name : list
+        List of paths to output mechanism files (.ck, .therm, .tran)
+    Examples
+    --------
+    >>> gas = cantera.Solution('gri30.cti')
+    >>> yaml2ck.write(gas)
+    [output_path / reduced_gri30.ck, output_path / reduced_gri30.therm,
+     output_path / reduced_gri30.tran]
+    """
+    if output_path:
+        if not isinstance(output_path, PurePath):
+            try:  # try to turn full path into pathlib Path
+                output_path = Path(output_path)
+            except:  # if previous fails, assume str is file name
+                output_path = Path.cwd() / output_path
+    else:
+        output_path = Path.cwd() / f"{solution.name}.ck"
+
+    if output_path.is_dir():
+        output_path = output_path / f"{solution.name}.ck"
+
+    if output_path.is_file():
+        output_path.unlink()
+
+    main_path = output_path.parents[0]
+    basename = output_path.stem
+    output_files = [output_path]
+
+    with open(output_path, "w") as mech_file:
+        # Write title block to file
+        if "description" in solution.input_header:
+            note = solution.input_header["description"]
+            mech_file.write(f"! {note}\n!\n")
+        else:
+            note = ""
+
+        if "! Chemkin file converted from Cantera solution object" not in note:
```

Comment:
For debugging purposes, it'd be nice to list the version of Cantera that produced the file, similar to `ck2yaml.py`.